docs/src/using-turing/interface.md

```diff
@@ -0,0 +1,321 @@
+
+We need the `MCMCChains.Chains` function, because at the end of sampling we need to be able to convert a vector of `<:AbstractTransitions` to an `MCMCChains` object.
+
+Lastly, we import the specific functions from `AbstractMCMC` that we need to overload. Because Metropolis-Hastings is so simple we only need a few functions/types:
+
+- `step!`
+- `AbstractSampler`
+- `AbstractTransition`
+- `transition_type`
+
+More complex inference methods may need to import additional functions. For now, let's focus on the code:
+
+```julia
+# Import specific functions and types to use or overload.
+import MCMCChains: Chains
+import AbstractMCMC: step!, AbstractSampler, AbstractTransition, transition_type, bundle_samples
+```
+
+### Sampler
+
+Let's begin our sampler definition by defining a sampler called `MetropolisHastings` which is a subtype of `AbstractSampler`. Correct typing is very important for proper interface implementation -- if you are missing a subtype, your method may not be dispatched to when you call `sample`.
+
+```julia
+# Define a sampler type.
+struct MetropolisHastings{T, D} <: AbstractSampler 
+    init_θ :: T
+    proposal :: D
+end
+
+# Default constructors.
+MetropolisHastings(init_θ::Real) = MetropolisHastings(init_θ, Normal(0,1))
+MetropolisHastings(init_θ::Vector{<:Real}) = MetropolisHastings(init_θ, MvNormal(length(init_θ),1))
+```
+
+Above, we have defined a sampler that stores the initial parameterization of the prior, and a distribution object from which proposals are drawn. You can have a struct that has no fields, and simply use it for dispatching onto the relevant functions, or you can store a large amount of state information in your sampler. 
+
+The general intuition for what to store in your sampler struct is that anything you may need to perform inference between samples but you don't want to store in an `AbstractTransition` should go into the sampler struct. It's the only way you can carry non-sample related state information between `step!` calls. 
+
+### Model
+
+Next, we need to have a model of some kind. A model is a struct that's a subtype of `AbstractModel` that contains whatever information is necessary to perform inference on your problem. In our case we want to know the mean and variance parameters for a standard Normal distribution, so we can keep our model to the log density of a Normal. 
+
+Note that we only have to do this because we are not yet integrating the sampler with Turing -- Turing has a very sophisticated modelling engine that removes the need to define custom model structs. 
+
+```julia
+# Define a model type. Stores the log density function.
+struct DensityModel{F<:Function} <: AbstractModel
+    ℓπ :: F
+end
+
+# Default density constructor.
+DensityModel(π::Function, data::T) where T = DensityModel{VariateForm, ValueSupport, T}(π, data)
```

Comment:
That seems wrong and does not match the `struct` definition above? Moreover, the static parameter could/should be removed (https://docs.julialang.org/en/v1/manual/style-guide/#Don't-use-unnecessary-static-parameters-1).